libtrellis/tools/ecpmulti.cpp

```diff
@@ -39,7 +40,10 @@ int main(int argc, char *argv[])
     using namespace Trellis;
     namespace po = boost::program_options;
 
-    std::string database_folder = TRELLIS_PREFIX "/share/trellis/database";
+    boost::filesystem::path executable_path = boost::dll::program_location().parent_path();
```

Comment:
What version of Boost did `boost::dll::program_location()` appear in? At a glance, looks like it doesn't feature in 1.58, which Ubuntu 16.04 provides and I'd still rather support for the time being?

Reply:
The dll module was added in Boost [1.61](https://www.boost.org/users/history/version_1_61_0.html).

There are a couple of solutions that would allow supporting Ubuntu 16.04.
1. Implement the equivalent lookup code in prjtrellis (e.g. https://github.com/cliffordwolf/icestorm/blob/master/icetime/iceutil.cc). It might be worth doing this as conditional on boost version, such that once support for 16.04 is dropped the code could also be dropped.
2. Just conditionally support portable execution depending on available boost version. Since portable binaries are likely to be built and shipped with libraries not from the host (thus having the ability to use a newer boost).

Let me know if either of these are an acceptable solution? or if I should look into other alternatives?

Reply:
I think that the former option is probably the best option for now, as you say we can then review it in the future depending on support priorities.

Reply:
I have reworked the second commit to include the requested changes. Please see the updated commit.